networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/ProposerSlashingValidatorTest.java

```diff
@@ -117,4 +119,26 @@ public void shouldIgnoreProposerSlashingForTheSameProposer() throws Exception {
     assertThat(proposerSlashingValidator.validate(slashing1)).isEqualTo(ACCEPT);
     assertThat(proposerSlashingValidator.validate(slashing2)).isEqualTo(IGNORE);
   }
+
+  @Test
+  public void shouldRejectProposerSlashingForTwoSignedHeadersWithSameMessageButDifferentSignature()
+      throws Exception {
+    beaconChainUtil.initializeStorage();
+    beaconChainUtil.createAndImportBlockAtSlot(6);
+    stateTransitionValidator = new ProposerSlashingStateTransitionValidator();
+    SignedBeaconBlockHeader signedBeaconBlockHeader =
+        dataStructureUtil.randomSignedBeaconBlockHeader();
+    ProposerSlashing slashing =
+        new ProposerSlashing(
+            new SignedBeaconBlockHeader(
+                signedBeaconBlockHeader.getMessage(), BLSSignature.random(100)),
+            signedBeaconBlockHeader);
```

Comment:
(nit) I'd declare the second header as a variable, and assert that the 2 headers are not equal, just so we're being really explicit and verifying our assumptions:

```suggestion
    final SignedBeaconBlockHeader header2 = new SignedBeaconBlockHeader(signedBeaconBlockHeader.getMessage(), BLSSignature.random(100);
    assertThat(header2).isNotEqualTo(signedBeaconBlockHeader);
    ProposerSlashing slashing =
        new ProposerSlashing(
            header2,
            signedBeaconBlockHeader);
```

Reply:
Done.